### PR TITLE
If playlist has more than 50 tracks it creates multiple download link…

### PR DESCRIPTION
**Pull request by Vindexus**
…s for it

### CODE REVIEW

File: src/js/playlist.js [hunk 1]
```diff
@@ -93,7 +93,25 @@ function transformPlaylist (obj) {
...
     }
-    obj.downloadUrl = endpoint + '/playlist/' + obj._id + '/download?' + objectToQueryString(opts)
+    if(obj.tracks.length < 50) {
+      obj.downloadUrl = endpoint + '/playlist/' + obj._id + '/download?' + objectToQueryString(opts)
+    }
+    else {
+      obj.downloadLinks = []
+      var frm = 1;
+      var to = 0;
+      for(var i = 1; (i-1) * 50 <= obj.tracks.length; i++) {
```
Comment (lostlevels) [inline src/js/playlist.js]: Shouldn't this be `<` ?

Comment (Vindexus) [inline src/js/playlist.js]: I... don't know.

Comment (Vindexus) [inline src/js/playlist.js]: Yes.

Comment (tmathews) [inline src/js/playlist.js]: This whole block needs work.

File: src/js/playlist.js [hunk 2]
```diff
@@ -97,16 +97,18 @@ function transformPlaylist (obj) {
...
-      obj.downloadLinks = []
-      var frm = 1;
-      var to = 0;
-      for(var i = 1; (i-1) * 50 < obj.tracks.length; i++) {
-        opts.page = i
-        frm = (i - 1) * 50 + 1
-        to = Math.min(obj.tracks.length, frm + 49)
+      obj.downloadLinks = [];
+      var tracksPerPage = 50;
+      var frm;
```
Comment (lostlevels) [inline src/js/playlist.js]: remove frm and to here.

File: src/js/playlist.js [hunk 3]
```diff
@@ -97,16 +97,18 @@ function transformPlaylist (obj) {
...
+      var to;
+      var numPages = Math.ceil(obj.tracks.length / tracksPerPage);
+      for(var page = 1; page <= numPages; page++) {
+        opts.page = page
+        frm = (page - 1) * tracksPerPage + 1
```
Comment (lostlevels) [inline src/js/playlist.js]: and put where they are first used here.

**Pull request merged**